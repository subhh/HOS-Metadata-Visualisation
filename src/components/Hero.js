import React, { useMemo } from 'react'
import { Switch, Route } from 'react-router-dom'
import styled from 'styled-components'
import GraphContainer from 'container/GraphContainer'
import SearchContainer from 'container/SearchContainer'
import Map from 'components/Map'
import Navigation from './Navigation'
import Legend from './Legend'
import ButtonLink from './ButtonLink'
import { shadow } from 'utils'

const Wrapper = styled.div`
  position: relative;
`

const Toggle = styled(ButtonLink)`
  position: absolute;
  right: 0;
  bottom: 0;
  margin: ${({ theme: { space } }) => `${space[5]}px ${space[4]}px`};
  ${shadow};
`

const Search = styled(SearchContainer)`
  position: absolute;
  top: 0;
  left: 0;
  margin: ${({ theme: { space } }) => `${space[5]}px ${space[4]}px`};
`

const Nav = styled(Navigation)`
  position: absolute;
  top: 0;
  right: 0;
  margin: ${({ theme: { space } }) => `${space[5]}px ${space[4]}px`};
`

const Hero = ({ width, height, targetHeight }) =>
  useMemo(() => {
    return (
      <Wrapper>
        <Switch>
          <Route
            path="/map"
            render={() => <Map width={width} height={height} />}
          />
          <Route
            path="/graph"
            render={() => (
              <Wrapper>
                <GraphContainer
                  width={width}
                  height={height}
                  simulationHeight={targetHeight}
                  debug={false}
                />
                <Switch>
                  <Route exact path="/graph">
                    <Toggle to="/graph/explore">Themen erkunden</Toggle>
                  </Route>
                  <Route exact path="/graph/explore">
                    <Toggle to="/graph">zur Ãœbersicht</Toggle>
                  </Route>
                </Switch>
              </Wrapper>
            )}
          />
        </Switch>
        <Legend />
        <Search />
        <Nav />
      </Wrapper>
    )
  }, [width, height, targetHeight])

export default Hero
