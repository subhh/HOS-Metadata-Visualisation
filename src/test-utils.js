import React, { useReducer } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { MemoryRouter } from 'react-router-dom'
import { StateContext, DispatchContext, reducer, initialState } from './state'
import theme from './theme'

const AllTheProviders = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState)

  return (
    <ThemeProvider theme={theme}>
      <StateContext.Provider value={state}>
        <DispatchContext.Provider value={dispatch}>
          {children}
        </DispatchContext.Provider>
      </StateContext.Provider>
    </ThemeProvider>
  )
}

const WithRouter = ({ children }) => (
  <MemoryRouter>
    <AllTheProviders>{children}</AllTheProviders>
  </MemoryRouter>
)

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

export const customRenderWithRouter = (ui, options) =>
  render(ui, {
    wrapper: WithRouter,
    ...options,
  })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }

export const mockAsync = data => {
  jest.spyOn(global, 'fetch').mockImplementation(url => {
    return Promise.resolve({
      json: () => Promise.resolve(data),
    })
  })
}

export const restoreAsyncMock = () => {
  global.fetch.mockRestore()
}
