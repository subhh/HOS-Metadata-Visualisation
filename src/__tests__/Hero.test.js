import React from 'react'
import Hero from 'components/HeroContainer'
import { customRenderWithRouter } from 'test-utils'
import { MemoryRouter } from 'react-router-dom'

it('<Hero /> without route sould match snapshot', () => {
  const { container } = customRenderWithRouter(<Hero />, null, '/graph')
  expect(container.firstChild).toMatchSnapshot()
})

it('<Hero /> should match snapshot on route `/graph`', () => {
  const { container } = customRenderWithRouter(
    <MemoryRouter initialEntries={['/graph']}>
      <Hero />
    </MemoryRouter>
  )
  expect(container.firstChild).toMatchSnapshot()
})

it('<Hero /> should match snapshot on route `/graph/explore`', () => {
  const { container } = customRenderWithRouter(
    <MemoryRouter initialEntries={['/graph/explore']}>
      <Hero />
    </MemoryRouter>
  )
  expect(container.firstChild).toMatchSnapshot()
})
