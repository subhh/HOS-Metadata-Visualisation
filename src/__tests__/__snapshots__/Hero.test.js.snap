// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> should match snapshot on route \`/graph/explore\` 1`] = `
<div
  class="sc-eHgmQL dpAtYH"
>
  <div
    class="sc-eHgmQL dpAtYH"
  >
    <div
      class="sc-bdVaJa sc-jzJRlG klRcDP"
      height="400"
      width="400"
    />
    <a
      class="sc-dxgOiQ hTjMfW sc-cvbbAY gcHweU"
      href="/graph"
    >
      <button
        class="sc-bdVaJa sc-EHOje sc-ckVGcZ kCPMKo"
        color="blueMediumDark"
        font-size="inherit"
        font-weight="bold"
      >
        zur Ãœbersicht
      </button>
    </a>
  </div>
  <div
    class="sc-bdVaJa sc-eNQAEJ hwlgxD"
    font-size="1"
  >
    <div
      class="sc-bdVaJa sc-hMqMXs sc-kkGfuU ljBZyT"
      data-testid="toggle"
      height="40"
      width="40"
    >
      <svg
        class="sc-hSdWYo lnHmQl"
        fill="blueMediumDark"
      >
        help.svg
      </svg>
    </div>
  </div>
  <div
    class="sc-jWBwVP jxrrWj"
  >
    <div
      class="sc-bdVaJa Search__SearchContainer-sc-18xh3im-0 hkYogB"
    >
      <div
        class="sc-bdVaJa Search__IconContainer-sc-18xh3im-2 Search__SearchIconContainer-sc-18xh3im-3 cBQBHZ"
        data-testid="submit"
        height="36"
        width="36"
      >
        <svg
          class="Search___StyledSearchSVG-sc-18xh3im-5 bSZqyx"
        >
          search.svg
        </svg>
      </div>
      <input
        class="Search__Input-sc-18xh3im-1 ShqGu"
        color="blueMediumDark"
        data-testid="input"
        font-size="1"
        height="36"
        opacity="0"
        placeholder="Suchbegriff eingeben"
        size="20"
        type="text"
        value=""
        width="36px"
      />
    </div>
  </div>
  <div
    class="sc-jKJlTe giNbvS sc-brqgnP fLbqjn"
  >
    <a
      class="sc-dxgOiQ hTjMfW"
      href="/map"
    >
      <button
        class="sc-bdVaJa sc-EHOje sc-ckVGcZ gKdOru"
        color="blueMediumDark"
        font-size="inherit"
        font-weight="bold"
      >
        Einrichtungen
      </button>
    </a>
    <a
      aria-current="page"
      class="sc-dxgOiQ hTjMfW active"
      href="/graph"
    >
      <button
        class="sc-bdVaJa sc-EHOje sc-ckVGcZ gKdOru"
        color="blueMediumDark"
        font-size="inherit"
        font-weight="bold"
      >
        Kollektionen
      </button>
    </a>
  </div>
</div>
`;

exports[`<Hero /> should match snapshot on route \`/graph\` 1`] = `
<div
  class="sc-eHgmQL dpAtYH"
>
  <div
    class="sc-eHgmQL dpAtYH"
  >
    <div
      class="sc-bdVaJa sc-jzJRlG klRcDP"
      height="400"
      width="400"
    />
    <a
      class="sc-dxgOiQ hTjMfW sc-cvbbAY gcHweU"
      href="/graph/explore"
    >
      <button
        class="sc-bdVaJa sc-EHOje sc-ckVGcZ kCPMKo"
        color="blueMediumDark"
        font-size="inherit"
        font-weight="bold"
      >
        Themen erkunden
      </button>
    </a>
  </div>
  <div
    class="sc-bdVaJa sc-eNQAEJ hwlgxD"
    font-size="1"
  >
    <div
      class="sc-bdVaJa sc-hMqMXs sc-kkGfuU ljBZyT"
      data-testid="toggle"
      height="40"
      width="40"
    >
      <svg
        class="sc-hSdWYo lnHmQl"
        fill="blueMediumDark"
      >
        help.svg
      </svg>
    </div>
  </div>
  <div
    class="sc-jWBwVP jxrrWj"
  >
    <div
      class="sc-bdVaJa Search__SearchContainer-sc-18xh3im-0 hkYogB"
    >
      <div
        class="sc-bdVaJa Search__IconContainer-sc-18xh3im-2 Search__SearchIconContainer-sc-18xh3im-3 cBQBHZ"
        data-testid="submit"
        height="36"
        width="36"
      >
        <svg
          class="Search___StyledSearchSVG-sc-18xh3im-5 bSZqyx"
        >
          search.svg
        </svg>
      </div>
      <input
        class="Search__Input-sc-18xh3im-1 ShqGu"
        color="blueMediumDark"
        data-testid="input"
        font-size="1"
        height="36"
        opacity="0"
        placeholder="Suchbegriff eingeben"
        size="20"
        type="text"
        value=""
        width="36px"
      />
    </div>
  </div>
  <div
    class="sc-jKJlTe giNbvS sc-brqgnP fLbqjn"
  >
    <a
      class="sc-dxgOiQ hTjMfW"
      href="/map"
    >
      <button
        class="sc-bdVaJa sc-EHOje sc-ckVGcZ gKdOru"
        color="blueMediumDark"
        font-size="inherit"
        font-weight="bold"
      >
        Einrichtungen
      </button>
    </a>
    <a
      aria-current="page"
      class="sc-dxgOiQ hTjMfW active"
      href="/graph"
    >
      <button
        class="sc-bdVaJa sc-EHOje sc-ckVGcZ gKdOru"
        color="blueMediumDark"
        font-size="inherit"
        font-weight="bold"
      >
        Kollektionen
      </button>
    </a>
  </div>
</div>
`;

exports[`<Hero /> without route sould match snapshot 1`] = `
<div
  class="sc-eHgmQL dpAtYH"
>
  <div
    class="sc-jWBwVP jxrrWj"
  >
    <div
      class="sc-bdVaJa Search__SearchContainer-sc-18xh3im-0 hkYogB"
    >
      <div
        class="sc-bdVaJa Search__IconContainer-sc-18xh3im-2 Search__SearchIconContainer-sc-18xh3im-3 cBQBHZ"
        data-testid="submit"
        height="36"
        width="36"
      >
        <svg
          class="Search___StyledSearchSVG-sc-18xh3im-5 bSZqyx"
        >
          search.svg
        </svg>
      </div>
      <input
        class="Search__Input-sc-18xh3im-1 ShqGu"
        color="blueMediumDark"
        data-testid="input"
        font-size="1"
        height="36"
        opacity="0"
        placeholder="Suchbegriff eingeben"
        size="20"
        type="text"
        value=""
        width="36px"
      />
    </div>
  </div>
  <div
    class="sc-jKJlTe giNbvS sc-brqgnP fLbqjn"
  >
    <a
      class="sc-dxgOiQ hTjMfW"
      href="/map"
    >
      <button
        class="sc-bdVaJa sc-EHOje sc-ckVGcZ gKdOru"
        color="blueMediumDark"
        font-size="inherit"
        font-weight="bold"
      >
        Einrichtungen
      </button>
    </a>
    <a
      class="sc-dxgOiQ hTjMfW"
      href="/graph"
    >
      <button
        class="sc-bdVaJa sc-EHOje sc-ckVGcZ gKdOru"
        color="blueMediumDark"
        font-size="inherit"
        font-weight="bold"
      >
        Kollektionen
      </button>
    </a>
  </div>
</div>
`;
